client/lib/analytics/utils.js

```diff
@@ -111,9 +107,9 @@ export function shouldSkipAds() {
  * @param {String} path The tracked path.
  * @returns {Boolean} If the report should null `blog_id`.
  */
-export const shouldReportOmitBlogId = path =>
-	!! path &&
-	some(
-		NO_BLOG_ID_PATHS,
-		noBlogIdPath => path === noBlogIdPath || path.indexOf( noBlogIdPath + '/' ) === 0
-	);
+export const shouldReportOmitBlogId = path => {
+	if ( ! path ) {
+		return true;
+	}
+	return ! SITE_FRAGMENT_REGEX.test( path );
```

Comment:
Wondering: is a regex test here better for performance reasons?
I'd find more `lodash.includes` more readable actually. 🤔 

Also, `\/:site(\/|$)` would fail to recognize "incorrect-ish" (but quite common AFAIK) site fragment variations such as `:site_id` or `:siteId`.

Reply:
Oh thanks I didn't realize we had that many variants. I'll add a few more test cases and update the code :)

Reply:
> is a regex test here better for performance reasons?

Mostly that it was easier for me to express more complex matching requirements, without looking for polyfills for endsWith. I think the complexity here shouldn't be too bad, and we'll be well covered with tests.

Reply:
Updated in feb9220